modules/examiner/jsx/examiner.js

```diff
@@ -0,0 +1,252 @@
+   * Additionaly add hiddenHeaders to global loris vairable
+   * for easy access by columnFormatter.
+   */
+  fetchData() {
+    const self = this;
+    $.ajax(this.props.DataURL, {
+      method: "GET",
+      dataType: 'json',
+      success: function(data) {
+        loris.hiddenHeaders = data.hiddenHeaders ? data.hiddenHeaders : [];
+        self.setState({
+          Data: data,
+          isLoaded: true
+        });
+      },
+      error: error => {
+        console.error(error);
+      }
+    });
+  }
+
+  pickElements(json, keys) {
+    let subset = {};
+    keys.forEach(function(key) {
+      if (json.hasOwnProperty(key)) {
+        subset[key] = json[key];
+      }
+    });
+    return subset;
+  }
+
+  setFormData(formElement, value) {
+    let formData = this.state.formData;
+    formData[formElement] = value;
+    this.setState({
+      formData: formData
+    });
+  }
+
+  updateFilter(filter) {
+    this.setState({filter});
+  }
+
+  resetFilters() {
+    this.refs.examinerFilter.clearFilter();
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    let formData = this.state.formData;
+    let formObject = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== "") {
+        formObject.append(key, formData[key]);
+      }
+    }
+    formObject.append("fire_away", "Add");
+    $.ajax({
+      type: 'POST',
+      url: loris.BaseURL + "/examiner/",
+      data: formObject,
+      cache: false,
+      contentType: false,
+      processData: false,
+      success: data => {
+        swal("Success!", "Examiner added.", "success");
+        this.fetchData();
+      },
+      error: error => {
+        console.error(error);
+        let message = error.responseText;
+        swal("Error!", message, "error");
+      }
+    });
+  }
+
+  render() {
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return (
+        <Loader/>
+      );
+    }
+
+    if (this.state.Data.form) {
+      // Make elements required
+      this.state.Data.form.addName.required = true;
+      this.state.Data.form.addSite.required = true;
+
+      // Set value in Add Examiner form
+      this.state.Data.form.addName.value = this.state.formData.addName;
+      this.state.Data.form.addRadiologist.value =
+          this.state.formData.addRadiologist;
+      this.state.Data.form.addSite.value = this.state.formData.addSite;
+
+      // Set onUserInput in Add Examiner form
+      this.state.Data.form.addName.onUserInput = this.setFormData;
+      this.state.Data.form.addRadiologist.onUserInput = this.setFormData;
+      this.state.Data.form.addSite.onUserInput = this.setFormData;
+    }
+
+    return (
+      <div>
+        <div id="forms" className="row">
+          <div id="filter-examiner" className="col-sm-12 col-md-7">
+            <FilterForm
+              Module="examiner"
+              name="examiner_filter"
+              id="examiner_filter_form"
+              ref="examinerFilter"
```

Comment:
no string refs plz

Reply:
i feel like refs can be used for now and a separate PR should be issued later to resolve all uses of refs. @driusan thoughts?

Reply:
keep the ref, just use the callback pattern. string refs are on their way to being deprecated